okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -691,21 +693,38 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
      * order than received.
      */
     fun applyAndAckSettings(clearPrevious: Boolean, settings: Settings) {
-      var delta = 0L
-      var streamsToNotify: Array<Http2Stream>? = null
+      var delta: Long
+      var streamsToNotify: Array<Http2Stream>?
```

Comment:
Should this change to List?

Reply:
In this case I’m trading off efficiency vs. readability. Probably the extra allocation of the list is why I avoided it initially. I don’t feel too strongly but I like treating this as “inner loop” code by default.